src/passes/function_passes/Utils/Utils.cpp

```diff
@@ -0,0 +1,29 @@
+#include "Utils.h"
+
+using namespace llvm;
+
+bool isPatchpoint(Function *fun) {
```

Comment:
Ah, let me clarify what I meant. Perhaps you can merge these two functions by having them return a bit field encoding the three possibilities?